### ISSUE TITLE
Quick impressions of frontier Chinese Lab models

### ISSUE THREAD

**Issue by noelsaw1** [HiQS-Labs/Needle-fork#54]
## TLDR

**Yes: enough for provisional impressions of four Chinese-lab models on our very narrow next-action quiz—not enough for a general frontier-model ranking.** GLM led this group at 50%, Qwen scored 40%, DeepSeek 36.7%, and Tencent 30%. The strongest simple baseline scored 46.7%; GLM's lead over it is just **one case**. None has established a dependable advantage or qualified for deployment.

These are retrospective descriptive grades, not new pass thresholds. We are grading agreement with the **next recorded coding action**, not coding skill, reasoning quality, the best action to recommend, or human acceptance. “Frontier” in this title describes the requested comparison, not a capability certification.

## Narrow-use-case scorecard

Same 30 cases / 30 distinct public issues, six possible labels, responses locked before scoring.

| Requested model / recorded route | Correct | Accuracy | Six-label macro-F1 | Quick impression |
|---|---:|---:|---:|---|
| ZCode GLM 5.3 Max; response self-reported GLM-5.3 | 15/30 | 50.0% | 0.346 | **Most promising in this group, still unproven.** One more correct case than phase-backoff; also highest macro-F1 of these four. |
| Qwen 3.8 Max; `qwen/qwen3.8-max-0902` | 12/30 | 40.0% | 0.333 | **Mixed.** Matched Markov-1 accuracy, two cases behind phase-backoff; macro-F1 was close to GLM's. |
| DeepSeek v4 Pro; `deepseek/deepseek-v4-pro` | 11/30 | 36.7% | 0.246 | **Limited predictive lift here.** Two cases above repeat-last, three below phase-backoff. |
| Tencent; `tencent/hy4-preview` | 9/30 | 30.0% | 0.283 | **Weakest accuracy in this group.** Matched repeat-last accuracy, but exceeded DeepSeek's macro-F1; not uniformly worst on every metric. |

Reference points: phase-backoff 14/30 (46.7%), Markov-1 12/30 (40.0%), repeat-last 9/30 (30.0%), majority 7/30 (23.3%). Across the full seven-model panel, Fable led at 18/30 (60.0%); Astra and Gemini Flash each scored 14/30. No statistical superiority claim.

## Concrete behavior, not personality judgments

All four produced 30 complete, unique, allowed-label predictions after collection/extraction. Label agreement is a separate question from successful delivery.

This sample contained only **one actual edit and zero Git actions**. GLM predicted edit nine times, Qwen seven, DeepSeek three, and Tencent eight; they predicted Git two, four, five, and six times respectively. That mismatch helps explain the accuracy results, but does not prove the models misunderstood software work. Our truncated context and mechanical action taxonomy may not align with a model's expectation of what ought to happen next. Raw-source audit #53 is still pending.

## Practical experience and cost

- **GLM:** collected through ZCode/AgentChorus. Runtime identity is self-reported; the requested **Max effort was not independently attested**. No comparable per-call cost receipt.
- **DeepSeek / Qwen / Tencent:** collected through Aider/OpenRouter. Selected routes were recorded, not independently verified backend identities. Aider reported **$0.04 / $0.14 / $0.07** respectively for these completed calls. These are historical single-call receipts, not current price quotes or a cost-performance benchmark.
- DeepSeek's initial launch failed on our local Aider YAML configuration before inference; the corrected retry succeeded. **Do not count that as a model failure.**
- Qwen/Tencent transcript line-wrapping required permissive JSON extraction without changing predictions. That is a delivery-format caveat, not demonstrated model instruction failure.
- **MiniMax M3: not run, therefore ungraded.** The proposed extra seat was not added.

## Limits and next step

Thirty reused development cases are enough for quick impressions, not stable vendor rankings. One answer changes accuracy by 3.33 percentage points. Label support is run_command 10, run_tests 9, read 7, search 3, edit 1, Git 0. Fixed six-label macro-F1 includes zero for unsupported Git for everyone. Harnesses and effort attestation differ; public-data pretraining exposure is unknown. These are not private workflow or user-preference measurements.

All seven panel members missed the same nine cases. The next useful step is **#53's bounded audit of those nine plus three controls**, not another model seat or training run. Preserve these original scores even if the audit motivates a separately versioned representation or mapping change. Prior gates and PR #43's hold remain unchanged.

Evidence: [scored panel and review](https://github.com/HiQS-Labs/Needle-fork/blob/2c7bd5c/TESTS-RESULTS/2026-09-12-next-action-panel/SUMMARY.md), [aggregate metrics](https://github.com/HiQS-Labs/Needle-fork/blob/2c7bd5c/TESTS-RESULTS/2026-09-12-next-action-panel/metrics.json), #52. Parent project arc: #1. Documentation-only impressions; no new model calls or experiment executed for this issue.

**Comment by noelsaw1** [comment]
Operator-authorized addition: run previously untested minimax/minimax-m3 once on the unchanged 30-case quiz through the existing Aider/OpenRouter ask-mode collector (600-second cap). Tencent hy4-preview already has a locked 9/30 result, so reuse it, no reroll. The coordinator has seen the key; the fresh MiniMax call will receive only unchanged quiz inputs and task instructions, no scores/peer answers or repository map. Lock the first complete answer before scoring; report this as a post-panel addition, preserve #52's seven-seat receipt and old gates. Success means 30 valid unique predictions plus a replayable same-case score, not promotion. #53 remains the next audit; this single user-requested call is an explicit exception to the earlier pause. Update this issue with the result and practical caveats.

**Comment by noelsaw1** [comment]
## Update: MiniMax tested; Tencent result reused

**MiniMax M3 scored 10/30 (33.3%). Tencent hy4-preview remains 9/30 (30.0%); it was already tested and was not rerun.** MiniMax is one case above repeat-last and four below phase-backoff. It does not change the recommendation to audit before more model calls or training.

This supersedes the original “MiniMax not run” status below; the original four-model impressions and #52's frozen seven-seat results are preserved as history. The operator explicitly requested this additional call after scores were public, so it is a **post-panel exploratory addition**, not another predeclared seat or a fresh validation set.

| Model | Correct / 30 | Accuracy | Six-label macro-F1 |
|---|---:|---:|---:|
| GLM 5.3 (Max effort unattested) | 15 | 50.0% | 0.346 |
| Qwen 3.8 Max | 12 | 40.0% | 0.333 |
| DeepSeek v4 Pro | 11 | 36.7% | 0.246 |
| **MiniMax M3 — newly tested** | **10** | **33.3%** | **0.241** |
| Tencent hy4-preview — retained result | 9 | 30.0% | 0.283 |
| Phase-backoff baseline | 14 | 46.7% | 0.254 |

**MiniMax impression: inexpensive successful delivery, weak next-action accuracy on this quiz.** Nine of its 30 predictions were Git, against zero Git targets. It got 4/7 reads and 5/10 run_command targets right, but only 1/9 run_tests and none of the three search or one edit targets. Ordinary action-change accuracy was 8/21 (38.1%). This describes observed label agreement, not general coding ability or whether its suggestions were sensible.

MiniMax correctly predicted one of the original nine all-seven misses. Eight cases were therefore missed by all eight models; at least one model was correct on 22/30. That is oracle coverage, **not a deployable ensemble score**. The original seven-seat 9-miss/21-covered figures remain correct for that original panel.

### Collection and verification

- Requested route: `openrouter/minimax/minimax-m3`, confirmed in the [OpenRouter listing](https://openrouter.ai/minimax/minimax-m3) and local Aider launch header. Backend identity remains independently unverified; the model itself reported “unverified.”
- Existing Aider 0.86.2 ask-mode collector; repository map/history restoration disabled, no files supplied to the model, no source retrieval. One call completed, one answer, zero failed workers, within the 600-second cap.
- Exact same frozen 30 input objects and six label definitions; no targets or peer answers in its packet. The coordinator knew the earlier answers; the fresh call did not receive them. Public-data pretraining contamination remains unknown.
- First complete response locked before scoring. All 30 predictions valid and unique. Aider line-wrapping required permissive JSON extraction in the disclosure string; predictions were unchanged.
- Aider reported **$0.0072** for this call/session. Historical receipt, not a guaranteed price or controlled speed/cost comparison.
- 441 non-slow tests passed, 6 skipped, 11 deselected. Original seven-seat scorer replay exactly matched the published receipt; new lock/transcript and quiz checks, existing malformed-input/wrong-prediction controls, independent confusion arithmetic and identical MiniMax replay passed. Credential contents checked absent from retained artifacts.
- Raw transcript, lock, quiz packet and replay script retained under ignored `data/next-action-panel-v1/`; reproduce locally with `.venv-mlx-spike/bin/python data/next-action-panel-v1/score-minimax.py`. No tracked source files, runtime, training or PR #43 changes.

### Next

Proceed with **#53's already scoped 12-event audit**: the original nine shared misses plus three controls. Preserve that selection even though MiniMax got one of the nine right. Check raw next calls and visible context, distinguish reproducible mapping defects from taxonomy boundaries, missing context and behavioral ambiguity, then choose one bounded follow-up. No manual user ratings required. The audit has not started.

<details>
<summary>Aggregate receipt and provenance hashes</summary>

```json
{
  "issue": 54,
  "status": "exploratory_post_panel_addition_no_promotion",
  "requested_route": "openrouter/minimax/minimax-m3",
  "model_identity_verified": false,
  "collection_cost_usd": 0.0072,
  "cases": 30,
  "issues": 30,
  "metrics": {
    "rows": 30,
    "correct": 10,
    "accuracy_pct": 33.333333333333336,
    "macro_f1": 0.24145299145299146,
    "labels": {
      "edit": {
        "support": 1,
        "predicted": 5,
        "correct": 0,
        "recall_pct": 0,
        "f1": 0
      },
      "git": {
        "support": 0,
        "predicted": 9,
        "correct": 0,
        "recall_pct": null,
        "f1": 0
      },
      "read": {
        "support": 7,
        "predicted": 6,
        "correct": 4,
        "recall_pct": 57.142857142857146,
        "f1": 0.6153846153846154
      },
      "run_command": {
        "support": 10,
        "predicted": 5,
        "correct": 5,
        "recall_pct": 50,
        "f1": 0.6666666666666666
      },
      "run_tests": {
        "support": 9,
        "predicted": 3,
        "correct": 1,
        "recall_pct": 11.11111111111111,
        "f1": 0.16666666666666666
      },
      "search": {
        "support": 3,
        "predicted": 2,
        "correct": 0,
        "recall_pct": 0,
        "f1": 0
      }
    },
    "change_rows": 21,
    "change_accuracy_pct": 38.095238095238095,
    "issues": 30,
    "issue_macro_accuracy_pct": 33.333333333333336,
    "permutation_null_expected_accuracy_pct": 14.444444444444445
  },
  "original_seven_all_wrong": 9,
  "original_shared_misses_corrected": 1,
  "all_eight_wrong": 8,
  "eight_seat_oracle_correct": 22,
  "tencent_reused_correct": 9,
  "strongest_baseline_correct": 14,
  "hashes": {
    "data/next-action-panel-v1/score-minimax.py": "b70beff3b0a9c3c9a0045aca832ff30a85c372113f23c0f02fbbb5656479bcd1",
    "data/next-action-panel-v1/minimax-locked.json": "9d1c5ee2aba735cec8a073b1e9ecea8449d8eca0eb268b9bd2b09230a15b825c",
    "data/next-action-panel-v1/consult-runs/panel-v1-minimax-233117/panel-v1-minimax.aider.md": "6364a58231d46efefee9bda57a40638f47c75513a2adfb3dd644191e65cba95b",
    "data/next-action-panel-v1/minimax-packet.md": "6dbd534397e8de61890e2970804c6859bde6e6e5c5b9864b75df186249a3a752",
    "TESTS-RESULTS/2026-09-12-next-action-panel/metrics.json": "72aa70b69721bd29b9c814e8173d2beb203e4720db1c50c2b4121fdbedc0b382"
  },
  "verification": "Frozen seven-seat replay identical; original hash/input/red controls passed; MiniMax lock matches raw final answer; same quiz; independent confusion arithmetic agrees."
}
```
</details>

**Comment by noelsaw1** [comment]
Audit update: #53 is now complete (https://github.com/HiQS-Labs/Needle-fork/blob/f225278/TESTS-RESULTS/2026-09-12-panel-source-audit/SUMMARY.md). Source alignment passed, but two concrete test-execution/metadata mapping defects were confirmed and filed as #56. This strengthens the caution that these impressions measure agreement with an imperfect mechanical taxonomy, not general coding ability. All five-model scores and the original seven-seat receipt stay unchanged; no retrospective regrading or promotion. Next is scoped #56 correction, not another model seat.
